plugins/bazaar/src/api.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import { createApiRef, IdentityApi } from '@backstage/core-plugin-api';
+import { BazaarProject, Status } from './types';
+
+export const getEntityRef = (entity: Entity) => {
+  return `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
+};
+
+export const bazaarApiRef = createApiRef<BazaarApi>({
+  id: 'bazaar',
+  description: 'Used to make requests towards the bazaar backend',
+});
+
+export interface BazaarApi {
+  updateMetadata(
+    baseUrl: string,
+    entity: Entity,
+    name: string,
+    announcement: string,
+    status: Status,
+  ): Promise<any>;
+
+  getMetadata(baseUrl: string, entity: Entity): Promise<any>;
+
+  getMemberCounts(
+    bazaarProjects: BazaarProject[],
+    baseUrl: string,
+  ): Promise<Map<string, number>>;
+
+  getMembers(baseUrl: string, entity: Entity): Promise<any>;
+
+  deleteMember(baseUrl: string, entity: Entity): Promise<void>;
+
+  deleteMembers(baseUrl: string, entity: Entity): Promise<void>;
+
+  addMember(baseUrl: string, entity: Entity): Promise<void>;
+
+  getEntities(baseUrl: string): Promise<any>;
+
+  deleteEntity(baseUrl: string, entity: Entity): Promise<void>;
+}
+
+export class BazaarClient implements BazaarApi {
+  private readonly identityApi: IdentityApi;
+
+  constructor(options: { identityApi: IdentityApi }) {
+    this.identityApi = options.identityApi;
+  }
+
+  async updateMetadata(
+    baseUrl: string,
```

Comment:
This is great! One more thing here though is that we can get rid of all the `baseUrl` passing in here for this client by using the `discoveryApi` and using that like we do in the scaffolder client https://github.com/backstage/backstage/blob/a111b43771a2a825802e4102930b818c4ba79337/plugins/scaffolder/src/api.ts#L95

Will remove the need to use the `configApi` at all in all the components.